ibind/client/ibkr_client.py

```diff
@@ -63,10 +87,392 @@ def __init__(
+        """
+
+        REQUEST_URL = "https://api.ibkr.com/v1/api/oauth/live_session_token"
+        REQUEST_METHOD = "POST"
+        ENCRYPTION_METHOD = "RSA-SHA256"
+
+        
+        load_dotenv()
+        config = configparser.ConfigParser()
+        config.read('D:\\git_repos\\oauth_env\\oauth_test.env')
+
+        consumer_key = config['consumer_key']['CONSUMER_KEY']
+        access_token_secret=config['access_token_secret']['ACCESS_TOKEN_SECRET']
+        encription_key_fp=config['keys']['ENCRYPTION_KEY_FP']
+        dh_prime_fp=config['Diffie_Hellman']['DH_PRIME_FP']
+        dh_generator=config['Diffie_Hellman']['DH_GENERATOR']
+        realm=config['realm']['REALM']
+
+        dh_random = self.generate_dh_random_bytes()
+        dh_prime=self.pem_to_dh_prime(pem_file_path=dh_prime_fp)
+        dh_challenge = self.generate_dh_challenge(
+            dh_prime=dh_prime,
+            dh_generator=dh_generator,
+            dh_random=dh_random,
+        )
+        prepend = self.calculate_live_session_token_prepend(
+            access_token_secret,
+            self.read_private_key(encription_key_fp),
+        )
+
+        # TO DO: call post and request from rest_client.py with additional parameters, instead of IBKR send_oauth_request function
+        extra_headers={"diffie_hellman_challenge": dh_challenge}
+
+        params={
+            'extra_headers':{"diffie_hellman_challenge": dh_challenge},
+            'prepend':prepend,
+            'signature_method':ENCRYPTION_METHOD,
+        }
+        response=self.post(path=REQUEST_URL, params = params, log = True)
+        # response = send_oauth_request(
+        #     request_method=REQUEST_METHOD,
+        #     request_url=REQUEST_URL,
+        #     oauth_token=access_token,
+        #     signature_method=ENCRYPTION_METHOD,
+        #     extra_headers={
+        #         "diffie_hellman_challenge": dh_challenge,
+        #     },
+        #     prepend=prepend,
+        # )
+
+        if not response.ok:
+            raise Exception(f"Live session token request failed: {response.text}")
+        
+        response_data = response.json()
+        lst_expires = response_data["live_session_token_expiration"]
+        dh_response = response_data["diffie_hellman_response"]
+        lst_signature = response_data["live_session_token_signature"]
+        live_session_token = self.calculate_live_session_token(
+            dh_prime=dh_prime,
+            dh_random_value=dh_random,
+            dh_response=dh_response,
+            prepend=prepend,
+        )
+        if not self.validate_live_session_token(
+            live_session_token=live_session_token,
+            live_session_token_signature=lst_signature,
+            consumer_key=consumer_key,
+        ):
+            raise Exception("Live session token validation failed.")
+        
+        return live_session_token, lst_expires
+    
+
+    
+    def get_headers(        
```

Comment:
Here's an idea:

Since in reality all the oauth additions are standalone functions that just call one another, with - as far as I can see - no persistent/temporary state that needs to be stored in a class, I think we should move these out of IbkrClient.

The contents of this method, and all the other methods added for oauth (`pem_to_dh_prime`, `generate_request_timestamp`, `read_private_key`, etc..) we can move to a separate module, let's say 'client/ibkr_oauth.py'. Then, this `IbkrClient.get_headers()` method would simply call an entrypoint function from that new module, let's say the function could be called `generate_headers`.

So, to summarise:
* We move all oauth additions to their own file
* We move the `get_headers()` contents to a new function in that file
* Here we simply call that new function and return whatever it produces

This way we improve their testability, separate the concerns and declutter the IbkrClient class. Open to suggestions if you can see some reasons for why it should be kept the way it is 👍

Reply:
I agree, I just hadn't got there yet!